### PR TITLE
CICD doc: Feature Branch instead of GitFlow

### PR DESCRIPTION
**Pull request by francardoso93**

### CODE REVIEW

File: cicd-workflows.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # CI/CD Workflows
 
-All Elastic Provider component repositories follow the [Gitflow Workflow strategy](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow). They can have any number of short-lived feature branches and two long-lived branches, which are: `dev` and `main`.
+All Elastic Provider component repositories follow the [Feature Branch Workflow strategy](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow). They can have any number of short-lived feature branches and two long-lived branches, which are:  `main` and `dev`. All PRs should be targetting `main`, [while `dev` branch is used as an optional way of validating big features](#dev-branch).
```
Comment (simone-sanfratello) [inline cicd-workflows.md:3]: `dev` branch should not exists at all, it's a legacy thing from git flow and does not belong to the "feature branch" flow

File: cicd-workflows.md [hunk 2]
```diff
@@ -25,8 +25,8 @@ Run Test => Run Lint => Build Artifact (Docker Image)
 
  This is triggered when a build workflow has finished running successfully for the long-lived branches. Each of these is associated with the environments that should be updated.
 
-- dev: dev (Automatic)
 - main: Staging (Automatic) and Prod (Requires manual approval)
+- dev: dev (Automatic)
```
Comment (simone-sanfratello) [inline cicd-workflows.md:29]: we should not mix flows, feature branch strategy doesnt have env branches

File: cicd-workflows.md [hunk 3]
```diff
@@ -81,10 +81,13 @@ There is no need of configuring any kind of access from GitHub, it stores zero l
 
 [This repo](https://github.com/elastic-ipfs/shared-workflows) stores generic workflows that are reused by several components.
 
+## `Dev` branch
```
Comment (simone-sanfratello) [inline cicd-workflows.md:84]: remove

**Comment by simone-sanfratello** [comment]
I think we should cover other operations: how to deploy in dev/staging/main, required reviews, deal with hotfix, provide examples
we should also set "owners" for each repo, people that are able to deploy, and make them consistent

**Review by simone-sanfratello — CHANGES_REQUESTED** [review]

File: cicd-workflows.md (hunk 1 above)
Comment (francardoso93) [reply cicd-workflows.md:3]: What other approach do you think it's ideal to validate/test features that will take longer to be deployed without polluting main branch?

**Comment by francardoso93** [comment]
> I think we should cover other operations: how to deploy in dev/staging/main, required reviews, deal with hotfix, provide examples we should also set "owners" for each repo, people that are able to deploy, and make them consistent

**how to deploy in dev/staging/main:** I thought I had covered that, but feel free to suggest how to improve the descriptions that currently exist about that.
**deal with hotfix**: There is no such thing in feature branch workflow
**provide examples**: Examples of.. ?
**we should also set "owners" for each repo, people that are able to deploy, and make them consistent**:  Does that affect docs? How would you put it?

**Comment by simone-sanfratello** [comment]
Can we at least remove the `dev` branch?

File: cicd-workflows.md (hunk 3 above)
Comment (francardoso93) [reply cicd-workflows.md]: Done

File: cicd-workflows.md (hunk 2 above)
Comment (francardoso93) [reply cicd-workflows.md]: Let's talk more about that one later, but removed for now

**Comment by francardoso93** [comment]
> Can we at least remove the dev branch?
Done :)

**Review by simone-sanfratello — APPROVED** [review]

**Pull request merged**